src/fxa-geodb.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// these are defaults, can be overloaded by configuring options
+var db = __dirname + '/../db/cities-db.mmdb';
+var db_backup = __dirname + '/../db/cities-db.mmdb-backup';
+var ERRORS = require('../lib/errors');
+var maxmind = require('maxmind');
+var Promise = require('bluebird');
+
+module.exports = function (options) {
+  'use strict';
+  options = options || {};
+  db = options.db || db;
+  db_backup = options.db_backup || db_backup;
+
+  return function (ip) {
+    return new Promise(function (resolve, reject) {
+      // check if ip is valid
+      if (!maxmind.validate(ip)) {
+        reject({
+          message: ERRORS.IS_INVALID
+        });
+      }
+
+      var city_lookup, city_data;
+      // ip is valid, try looking it up
+      // the nested try..catch is to ensure that
+      // we always have at least one valid database check
+      // this can help when we have `db` as paid and
+      // `db_backup` as free version or when `db` fails
+      // to load for some reason
+      try {
+        city_lookup = maxmind.open(db);
+        city_data = city_lookup.get(ip);
+      } catch (err) {
+        // if it failed with primary database,
+        // try with backup database
+        try {
+          city_lookup = maxmind.open(db_backup);
```

Comment:
I see a couple of problems here. 

Keep in mind how and where this library is going to be used - the lookup function has the potential to be called many thousands of times an hour. Opening and reading the DB file synchronously for every query is going to be a performance bottleneck. Instead, open the DB on startup, keep it open for all queries. If the main DB file isn't available/valid, open the backup. If neither is usable, throw an error.

Next, I'm not sure there is any real value in querying the backup database if the info is not found in the primary database. I'd just stick with one for now.